cascade-merge/action.yml

```diff
@@ -0,0 +1,162 @@
+        git config user.email "github-actions@github.com"
+      shell: bash
+
+    - name: Create PR
+      id: create_prs
+      run: |
+        # Enables the shell to print all executed commands and exit if any command fails
+        # set -ex
+        
+        # A function to compare semantic version numbers. Echoes 1 if the first is greater, 2 if the second is greater, 0 if they are equal
+        compare_versions() {
+            IFS='.' read -ra VERSION1 <<< "$1"
+            IFS='.' read -ra VERSION2 <<< "$2"
+        
+            for i in {0..2}; do
+                PART1=${VERSION1[i]:-0}
+                PART2=${VERSION2[i]:-0}
+                if (( PART1 > PART2 )); then
+                    echo 1
+                    return
+                elif (( PART1 < PART2 )); then
+                    echo 2
+                    return
+                fi
+            done
+            echo 0
+        }
+        
+        # beautify the message by coloring it to a green color
+        function echo_green {
+            echo -e "\e[32m$1\e[0m"
+        }
+        
+        PREFIX="auto-cascade-merge"
+        ORIGINAL_BRANCH="${GITHUB_REF#refs/heads/}"
+        # backport target branch pattern is harcoded here as `release/`
+        BACKPORT_TARGET_BRANCH_PATTERN="release/"
+        # Remove the branch pattern prefix from the original branch name to get the original version
+        ORIGINAL_VERSION="${ORIGINAL_BRANCH#"$BACKPORT_TARGET_BRANCH_PATTERN"}"
+
+        # List and sort (in reverse, sorting in semantic version way) all the branches that match the pattern
+        RELEASE_VERSIONS=$(git branch -r | grep -oP "origin/${BACKPORT_TARGET_BRANCH_PATTERN}\\K.*" | sort -Vr)
+        TARGET_VERSIONS="${{ inputs.default-branch }}"
+        
+        # Loop through all release versions
+        for VERSION in $RELEASE_VERSIONS; do
+          echo "Comparing versions $VERSION and $ORIGINAL_VERSION"
+          compare_versions "$VERSION" "$ORIGINAL_VERSION"
+          RESULT=$(compare_versions "$VERSION" "$ORIGINAL_VERSION")
+          # If the version in the loop is greater than the original version, add it to the target versions
+          if [[ $RESULT -eq 1 ]]; then
+            TARGET_VERSIONS="$TARGET_VERSIONS $VERSION"
+          fi
+        done
+        
+        NEW_PR_URLS=""
+        
+        # Loop through all target versions
+        for TARGET_VERSION in $TARGET_VERSIONS; do
+          if [[ $TARGET_VERSION == "${{ inputs.default-branch }}" ]]; then
+            TARGET_BRANCH="${{ inputs.default-branch }}"
+          else
+            TARGET_BRANCH="$BACKPORT_TARGET_BRANCH_PATTERN$TARGET_VERSION"
+          fi
+        
+          BRANCH_NAME="$PREFIX-${ORIGINAL_BRANCH}-to-$TARGET_BRANCH"
+
+          # Get the PR number of the existing PR (if any) that has the same base and head
+          PR_NUMBER=$(gh pr list --base $TARGET_BRANCH --head $BRANCH_NAME --json number --jq '.[0].number')
```

Comment:
This makes the hypothesis that github cli is installed on the runner.
Is it ensured? Otherwise can we add some steps that install and configure it?

Reply:
on github runners, `ubunut-latest` is installed by default.
am checking on internal shared runners if it's the case

Reply:
it's not installed [indeed on shared runners](https://github.com/LedgerHQ/cascade-merge-experiment/actions/runs/5412949360/jobs/9837780908)

I will work on installing it if it's not installed

Reply:
fixed by integrating a github action to install gh-cli